astropy/io/fits/scripts/fitsinfo.py

```diff
@@ -0,0 +1,64 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+``fitsinfo`` is a command-line script based on astropy.io.fits for
+printing a summary of the HDUs in one or more FITS files(s) to the
+standard output.
+
+Example usage of ``fitsinfo``:
+
+1. Print a summary of the HDUs in a FITS file::
+
+    $ fitsinfo filename.fits
+
+    Filename: filename.fits
+    No.    Name         Type      Cards   Dimensions   Format
+    0    PRIMARY     PrimaryHDU     138   ()
+    1    SCI         ImageHDU        61   (800, 800)   int16
+    2    SCI         ImageHDU        61   (800, 800)   int16
+    3    SCI         ImageHDU        61   (800, 800)   int16
+    4    SCI         ImageHDU        61   (800, 800)   int16
+
+2. Print a summary of HDUs of all the FITS files in the current directory::
+
+    $ fitsinfo *.fits
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import argparse
+import astropy.io.fits as fits
+from astropy import log
+
+
+def fitsinfo(filename):
+    """
+    Print a summary of the HDUs in a FITS file.
+
+    Parameters
+    ----------
+    filename : str
+        The path to a FITS file.
+    """
+
+    try:
+        hdulist = fits.open(filename)
+        hdulist.info()
+        hdulist.close()
+    except IOError as e:
+        log.error(str(e))
```

Comment:
Will this log message necessarily be printed?  I'm not sure, but now that I think about it, for the purposes of a command-line script, you may have to manually adjust the log-levels and such to ensure this actually outputs errors to the terminal.  It should also give a better error message than just printing the `str()` of the exception, which may not often be useful.